extensions/procfs/CpuStat.cpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CpuStat.h"
+
+using namespace std::literals::chrono_literals;
+
+namespace org::apache::nifi::minifi::extensions::procfs {
+
+std::optional<CpuStatData> CpuStatData::parseCpuStatLine(std::istream& iss) {
+  CpuStatData data;
+  iss >> data.user_ >> data.nice_ >> data.system_ >> data.idle_ >> data.io_wait_ >> data.irq_ >> data.soft_irq_ >> data.steal_ >> data.guest_ >> data.guest_nice_;
+  if (iss.fail())
+    return std::nullopt;
+  data.user_ -= data.guest_;  // Guest time is already accounted in usertime
+  data.nice_ -= data.guest_nice_;
```

Comment:
Thats an excellent question. I did this because I wanted to reproduce what htop reports.
https://github.com/htop-dev/htop/blob/main/linux/LinuxProcessList.c#L1964-L1965

Reply:
Yeah, both ways are reasonable.  But since this is a ProcFsMonitor, not an HTopMonitor, I think we should either output the same numbers as in `/proc/stat`, or document the difference from `/proc/stat` in PROCESSORS.md.

Reply:
I see your point, changed it in https://github.com/apache/nifi-minifi-cpp/pull/1152/commits/e4ba2c310d44114a90d4b4d1830a99aa63a14b75